android/src/main/java/com/brentvatne/exoplayer/ExoPlayerView.kt

```diff
@@ -0,0 +1,317 @@
+    private var player: ExoPlayer? = null
+    private var layoutParams: ViewGroup.LayoutParams = ViewGroup.LayoutParams(
+        ViewGroup.LayoutParams.MATCH_PARENT,
+        ViewGroup.LayoutParams.MATCH_PARENT
+    )
+    private var adOverlayFrameLayout: FrameLayout
+    val isPlaying: Boolean
+        get() = player != null && player?.isPlaying == true
+
+    @ViewType.ViewType
+    private var viewType = ViewType.VIEW_TYPE_SURFACE
+    private var hideShutterView = false
+
+    init {
+        componentListener = ComponentListener()
+
+        val aspectRatioParams = LayoutParams(
+            LayoutParams.MATCH_PARENT,
+            LayoutParams.MATCH_PARENT
+        )
+        aspectRatioParams.gravity = Gravity.CENTER
+        layout = AspectRatioFrameLayout(context)
+        layout.layoutParams = aspectRatioParams
+
+        shutterView = View(context)
+        shutterView.layoutParams = layoutParams
+        shutterView.setBackgroundColor(ContextCompat.getColor(context, android.R.color.black))
+
+        subtitleLayout = SubtitleView(context)
+        subtitleLayout.layoutParams = layoutParams
+        subtitleLayout.setUserDefaultStyle()
+        subtitleLayout.setUserDefaultTextSize()
+
+        updateSurfaceView(viewType)
+
+        adOverlayFrameLayout = FrameLayout(context)
+
+        layout.addView(shutterView, 1, layoutParams)
+        layout.addView(adOverlayFrameLayout, 2, layoutParams)
+
+        addViewInLayout(layout, 0, aspectRatioParams)
+        addViewInLayout(subtitleLayout, 1, layoutParams)
+    }
+
+    private fun clearVideoView() {
+        when (val view = surfaceView) {
+            is TextureView -> player?.clearVideoTextureView(view)
+
+            is SurfaceView -> player?.clearVideoSurfaceView(view)
+
+            else -> {
+                Log.w(
+                    "clearVideoView",
+                    "Unexpected surfaceView type: ${surfaceView?.javaClass?.name}"
+                )
+            }
+        }
+    }
+
+    private fun setVideoView() {
+        when (val view = surfaceView) {
+            is TextureView -> player?.setVideoTextureView(view)
+
+            is SurfaceView -> player?.setVideoSurfaceView(view)
+
+            else -> {
+                Log.w(
+                    "setVideoView",
+                    "Unexpected surfaceView type: ${surfaceView?.javaClass?.name}"
+                )
+            }
+        }
+    }
+
+    fun setSubtitleStyle(style: SubtitleStyle) {
+        // ensure we reset subtitle style before reapplying it
+        subtitleLayout.setUserDefaultStyle()
+        subtitleLayout.setUserDefaultTextSize()
+
+        if (style.fontSize > 0) {
+            subtitleLayout.setFixedTextSize(TypedValue.COMPLEX_UNIT_SP, style.fontSize.toFloat())
+        }
+        subtitleLayout.setPadding(
+            style.paddingLeft,
+            style.paddingTop,
+            style.paddingTop,
+            style.paddingBottom
+        )
+        if (style.opacity.toInt() != 0) {
```

Comment:
style.opacity is a float If I remenber well, why adding toInt() ?

Reply:
Thanks for good points. It's just fault. I applied your suggestion. [b99fe2a](https://github.com/TheWidlarzGroup/react-native-video/pull/4038/commits/b99fe2ab3a657feac816fc1a2af4b5152790c663)